lib/substitutions/include/substitutions/substitutions_v2.h

```diff
@@ -101,15 +101,19 @@ struct ParallelTensorPattern {
 };
 
 struct SubstitutionPattern {
-  OperatorPattern at(utils::Node) const;
-  ParallelTensorPattern at(PatternEdge) const;
+  OperatorPattern at(Node) const;
+  ParallelTensorPattern at(OpenMultiDiEdge) const;
 
-  std::unique_ptr<IMultiDiGraphPattern> graph;
-  utils::bidict<utils::Node, OperatorPattern> node_map;
-  utils::bidict<PatternEdge, ParallelTensorPattern> edge_map;
+  std::unique_ptr<IOpenMultiDiGraph> graph;
```

Comment:
`I*` interfaces generally shouldn't be used except in `utils` (there are some exceptions, but I don't think this is one)